test-data/write-test-data.pl

```diff
@@ -367,83 +368,126 @@ sub write_invalid_node_count {
+        fr      => 'utf8_string',
+        ja      => 'utf8_string',
+        'pt-BR' => 'utf8_string',
+        ru      => 'utf8_string',
+        'zh-CN' => 'utf8_string',
+
+        # production
         accuracy_radius                => 'uint16',
         autonomous_system_number       => 'uint32',
         autonomous_system_organization => 'utf8_string',
-        cellular                       => 'uint16',
+        average_income                 => 'uint32',
         city                           => 'map',
+        code                           => 'utf8_string',
         confidence                     => 'uint16',
+        connection_type                => 'utf8_string',
         continent                      => 'map',
         country                        => 'map',
+        domain                         => 'utf8_string',
         geoname_id                     => 'uint32',
+        is_anonymous                   => 'boolean',
+        is_anonymous_proxy             => 'boolean',
+        is_anonymous_vpn               => 'boolean',
+        is_hosting_provider            => 'boolean',
+        is_legitimate_proxy            => 'boolean',
+        is_public_proxy                => 'boolean',
+        is_satellite_provider          => 'boolean',
+        is_tor_exit_node               => 'boolean',
+        iso_code                       => 'utf8_string',
         isp                            => 'utf8_string',
         latitude                       => 'double',
         location                       => 'map',
         longitude                      => 'double',
         metro_code                     => 'uint16',
         names                          => 'map',
         organization                   => 'utf8_string',
+        population_density             => 'uint32',
         postal                         => 'map',
         registered_country             => 'map',
         represented_country            => 'map',
         subdivisions                   => [ 'array', 'map' ],
+        time_zone                      => 'utf8_string',
         traits                         => 'map',
-    );
+        traits                         => 'map',
+        type                           => 'utf8_string',
+        user_type                      => 'utf8_string',
+
+        # for testing only
+        foo       => 'utf8_string',
+        bar       => 'utf8_string',
+        buzz      => 'utf8_string',
+        our_value => 'utf8_string',
+    };
+
+    my $callback = sub {
+        my $key = shift;
 
-    my $type_cb = sub {
-        return 'boolean' if $_[0] =~ /^is_/;
-        return $type_map{ $_[0] } // 'utf8_string';
+        return $map->{$key} || die <<"ERROR";
+Unknown tree key '$key'.
+
+The universal_map_key_type_callback doesn't know what type to use for the passed
+key.  If you are adding a new key that will be used in a frozen tree / mmdb then
+you should update the mapping in MM::GeoIP2::Util to provide a type for that
```

Comment:
I'd prefer something we can just cut and paste into this public code rather than having to maintain a bunch of differences. I am not sure anyone cares that this is mentioned.